mcsema/BC/Instruction.cpp

```diff
@@ -41,6 +42,50 @@
 
 #include "mcsema/CFG/CFG.h"
 
+namespace {
+
+// Load the address of a register.
+static llvm::Value *LoadRegAddress(llvm::BasicBlock *block,
+                                   std::string reg_name) {
+  return new llvm::LoadInst(
+      remill::FindVarInFunction(block->getParent(), reg_name), "", block);
+}
+
+// Load the value of a register.
+static llvm::Value *LoadRegValue(llvm::BasicBlock *block,
+                                 std::string reg_name) {
+  return new llvm::LoadInst(LoadRegAddress(block, reg_name), "", block);
+}
+
+static llvm::Value *LoadAddressRegValOrZero(llvm::BasicBlock *block,
+                                         const remill::Operand::Register &reg,
+                                         llvm::ConstantInt *zero) {
+  if (reg.name.empty()) {
+    return zero;
+  }
+
+  auto value = LoadRegValue(block, reg.name);
+  auto value_type = llvm::dyn_cast_or_null<llvm::IntegerType>(value->getType());
```

Comment:
I think you want a `dyn_cast` and not `dyn_cast_or_null`. All `llvm::Value`s will have a `getType()`, i.e. that will never return `nullptr`.